### PR TITLE
Forward client IP to IAM

### PR DESCRIPTION
**Pull request by sajuptpm**
InfoSec requirement to Forward client IP to IAM.This patch read
X-Forwarded-For from request headers and add it in auth
request headers.

### CODE REVIEW

**Comment by sajuptpm** [comment]
warn log in added only to validate the changes after deployment. We can remove this log later.

File: keystonemiddleware/auth_token/__init__.py [hunk 1]
```diff
@@ -772,6 +777,14 @@ def _get_user_token_from_header(self, env):
         else:
             raise exc.InvalidToken(_('Unable to find token in headers'))
 
+    def _get_client_ip(self, env):
+        ips = self._get_header(env, 'X-Forwarded-For')
+        if ips:
+            return ips.split(',')[0]
```
Comment (gdpak) [inline keystonemiddleware/auth_token/__init__.py:784]: if there is no 'X-forwarded-For' header in request then we should not include NULL IP in the header . We can just avoid filling this header in this case

File: keystonemiddleware/auth_token/__init__.py [hunk 2]
```diff
@@ -672,7 +672,12 @@ def _fmt_msg(env):
                 self._LOG.info(_LI('Invalid user token - rejecting request'))
                 return self._reject_request(env, start_response)
 
-            headers = {'X-Auth-Token' : user_token}
+            client_ip = self._get_client_ip(env)
+            request_id = self._get_request_id_from_header(env)
+            self._LOG.warn(_LW('Client IP of request:{request_id} is {client_ip}'.\
+                                format(request_id=request_id, client_ip=client_ip)))
+            headers = {'X-Auth-Token' : user_token,
+                        'X-Forwarded-For': client_ip}
```
Comment (gdpak) [inline keystonemiddleware/auth_token/__init__.py]: do not fil null client ip if there is no 'x-forwarded-for' attribute in header

File: keystonemiddleware/auth_token/__init__.py (hunk 1 above)
Comment (sajuptpm) [reply keystonemiddleware/auth_token/__init__.py:784]: done

**Pull request merged**